src/sentry/snuba/metrics/fields/base.py

```diff
@@ -364,13 +503,16 @@ def generate_default_null_values(self):
     def generate_available_operations(self):
         return []
 
-    def run_post_query_function(self, data, idx=None):
+    def run_post_query_function(self, data, query_definition: QueryDefinition, idx=None):
```

Comment:
I have to add this parameter such that we can access histogram opts. since some fields are instantiated as singletons, there's no other way than to pass it as params

Reply:
I feel like the histogram parameters should be passed to the `op` itself, i.e. `histogram(metric_name, from, to, buckets)`. Then you would be able to query multiple histograms with different parameters in one request, e.g. `?field=histogram(slow_thing, 0, 10000, 250)&field=histogram(fast_thing, 0, 100, 250)`.

Reply:
we had to decide against that as it would mean writing a proper parser for fields. I also don't think this will be hard to support in the future